[ICE]: unexpected `def_kind` in `codegen_fn_attrs`: Const

<!--
[31mICE[0m: Rustc ./a.rs '' 'thread 'rustc' ($TID) panicked at compiler/rustc_codegen_ssa/src/codegen_attrs.rs:558:9: 'unexpected `def_kind` in `codegen_fn_attrs`: Const'', 'thread 'rustc' ($TID) panicked at compiler/rustc_codegen_ssa/src/codegen_attrs.rs:558:9: 'unexpected `def_kind` in `codegen_fn_attrs`: Const''
File: /tmp/im/a.rs
-->

auto-reduced (treereduce-rust):
````rust
#[target_feature(enable = "sse2")]
const fn sse2() {}

const SSE2_ONLY: () = unsafe {
    let sse2_ptr: unsafe fn() = sse2;
};

````

<details><summary><strong>original code</strong></summary>
<p>

original:
````rust
// Tests the new rules added by RFC 2396, including:
// - applying `#[target_feature]` to safe functions is allowed
// - calling functions with `#[target_feature]` is allowed in
//   functions which have (at least) the same features
// - calling functions with `#[target_feature]` is allowed in
//   unsafe contexts
// - functions with `#[target_feature]` can coerce to unsafe fn pointers

//@ check-pass
//@ only-x86_64

#[target_feature(enable = "sse2")]
const fn sse2() {}

#[target_feature(cfg = "avx")]
const SSE2_ONLY: () = unsafe {
    if cfg!(target_feature = "sse2") {
        unsafe {
            cfg();
            #[target_feature(enable = "sse2")];
        }
    }
    let sse2_ptr: unsafe fn() = sse2;
};

#[target_feature(enable = "sse2")]
fn also_sse2() {
    sse2();
}

#[target_feature(enable = "sse2")]
#[target_feature(enable = "avx")]
fn sse2_and_avx() {
    sse2();
}

struct Foo;

impl Foo {
    #[target_feature(enable = "sse2")]
    fn sse2(&self) {
        sse2();
    }
}

fn main() {
    if cfg!(target_feature = "sse2") {
        unsafe {
            sse2();
            Foo.sse2();
        }
    }
    let sse2_ptr: unsafe fn() = sse2;
}

````
</p>
</details>

Version information
````
rustc 1.95.0-nightly (13c38730d 2026-02-08)
binary: rustc
commit-hash: 13c38730d981289cc7ae4cc109fd7756bf83ee67
commit-date: 2026-02-08
host: x86_64-unknown-linux-gnu
release: 1.95.0-nightly
LLVM version: 22.1.0

````

Possibly related line of code:
https://github.com/rust-lang/rust/blob/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_codegen_ssa/src/codegen_attrs.rs#L552-L564

Command:
`/home/matthias/.rustup/toolchains/alt-master/bin/rustc `

<details><summary><strong>Program output</strong></summary>
<p>

```
error[E0601]: `main` function not found in crate `mvce`
 --> /tmp/icemaker_global_tempdir.9u07a71lJDFq/rustc_testrunner_tmpdir_reporting.InKgOnyXc7YJ/mvce.rs:6:3
  |
6 | };
  |   ^ consider adding a `main` function to `/tmp/icemaker_global_tempdir.9u07a71lJDFq/rustc_testrunner_tmpdir_reporting.InKgOnyXc7YJ/mvce.rs`


thread 'rustc' (3975334) panicked at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_codegen_ssa/src/codegen_attrs.rs:558:9:
unexpected `def_kind` in `codegen_fn_attrs`: Const
stack backtrace:
   0:     0x7fcf674e44d0 - std[64b2b1c9aa015027]::backtrace_rs::backtrace::libunwind::trace
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/../../backtrace/src/backtrace/libunwind.rs:117:9
   1:     0x7fcf674e44d0 - std[64b2b1c9aa015027]::backtrace_rs::backtrace::trace_unsynchronized::<std[64b2b1c9aa015027]::sys::backtrace::_print_fmt::{closure#1}>
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/../../backtrace/src/backtrace/mod.rs:66:14
   2:     0x7fcf674e44d0 - std[64b2b1c9aa015027]::sys::backtrace::_print_fmt
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/sys/backtrace.rs:74:9
   3:     0x7fcf674e44d0 - <<std[64b2b1c9aa015027]::sys::backtrace::BacktraceLock>::print::DisplayBacktrace as core[2d39e5c12c705ebb]::fmt::Display>::fmt
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/sys/backtrace.rs:44:26
   4:     0x7fcf62b41c1e - <core[2d39e5c12c705ebb]::fmt::rt::Argument>::fmt
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/core/src/fmt/rt.rs:152:76
   5:     0x7fcf62b41c1e - core[2d39e5c12c705ebb]::fmt::write
   6:     0x7fcf674fc94c - std[64b2b1c9aa015027]::io::default_write_fmt::<std[64b2b1c9aa015027]::sys::stdio::unix::Stderr>
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/io/mod.rs:639:11
   7:     0x7fcf674fc94c - <std[64b2b1c9aa015027]::sys::stdio::unix::Stderr as std[64b2b1c9aa015027]::io::Write>::write_fmt
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/io/mod.rs:1994:13
   8:     0x7fcf674a8ff6 - <std[64b2b1c9aa015027]::sys::backtrace::BacktraceLock>::print
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/sys/backtrace.rs:47:9
   9:     0x7fcf674a8ff6 - std[64b2b1c9aa015027]::panicking::default_hook::{closure#0}
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:292:27
  10:     0x7fcf674d3161 - std[64b2b1c9aa015027]::panicking::default_hook
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:319:9
  11:     0x7fcf63f3e346 - <alloc[21b5fb6b1b472667]::boxed::Box<dyn for<'a, 'b> core[2d39e5c12c705ebb]::ops::function::Fn<(&'a std[64b2b1c9aa015027]::panic::PanicHookInfo<'b>,), Output = ()> + core[2d39e5c12c705ebb]::marker::Sync + core[2d39e5c12c705ebb]::marker::Send> as core[2d39e5c12c705ebb]::ops::function::Fn<(&std[64b2b1c9aa015027]::panic::PanicHookInfo,)>>::call
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/alloc/src/boxed.rs:2220:9
  12:     0x7fcf63f3e346 - rustc_driver_impl[ecb1fe8f07c57781]::install_ice_hook::{closure#1}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_driver_impl/src/lib.rs:1484:17
  13:     0x7fcf63f3e346 - <alloc[21b5fb6b1b472667]::boxed::Box<rustc_driver_impl[ecb1fe8f07c57781]::install_ice_hook::{closure#1}> as core[2d39e5c12c705ebb]::ops::function::Fn<(&dyn for<'a, 'b> core[2d39e5c12c705ebb]::ops::function::Fn<(&'a std[64b2b1c9aa015027]::panic::PanicHookInfo<'b>,), Output = ()> + core[2d39e5c12c705ebb]::marker::Sync + core[2d39e5c12c705ebb]::marker::Send, &std[64b2b1c9aa015027]::panic::PanicHookInfo)>>::call
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/alloc/src/boxed.rs:2220:9
  14:     0x7fcf63f3e346 - std[64b2b1c9aa015027]::panicking::update_hook::<alloc[21b5fb6b1b472667]::boxed::Box<rustc_driver_impl[ecb1fe8f07c57781]::install_ice_hook::{closure#1}>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:235:47
  15:     0x7fcf674d34fa - <alloc[21b5fb6b1b472667]::boxed::Box<dyn for<'a, 'b> core[2d39e5c12c705ebb]::ops::function::Fn<(&'a std[64b2b1c9aa015027]::panic::PanicHookInfo<'b>,), Output = ()> + core[2d39e5c12c705ebb]::marker::Sync + core[2d39e5c12c705ebb]::marker::Send> as core[2d39e5c12c705ebb]::ops::function::Fn<(&std[64b2b1c9aa015027]::panic::PanicHookInfo,)>>::call
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/alloc/src/boxed.rs:2220:9
  16:     0x7fcf674d34fa - std[64b2b1c9aa015027]::panicking::panic_with_hook
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:833:13
  17:     0x7fcf674a90b8 - std[64b2b1c9aa015027]::panicking::panic_handler::{closure#0}
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:698:13
  18:     0x7fcf67494b59 - std[64b2b1c9aa015027]::sys::backtrace::__rust_end_short_backtrace::<std[64b2b1c9aa015027]::panicking::panic_handler::{closure#0}, !>
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/sys/backtrace.rs:182:18
  19:     0x7fcf674ab00d - __rustc[25666c693a360600]::rust_begin_unwind
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:689:5
  20:     0x7fcf62b4264c - core[2d39e5c12c705ebb]::panicking::panic_fmt
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/core/src/panicking.rs:80:14
  21:     0x7fcf63abc2a8 - rustc_codegen_ssa[5f2b84ab5877956e]::codegen_attrs::codegen_fn_attrs
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_codegen_ssa/src/codegen_attrs.rs:558:9
  22:     0x7fcf6643708e - rustc_query_impl[d22e6f0bf5601a23]::query_impl::codegen_fn_attrs::compute_fn::__rust_begin_short_backtrace
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_impl/src/plumbing.rs:294:13
  23:     0x7fcf6624a07f - <rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}> as rustc_query_system[42b0665e8e4da0bf]::query::dispatcher::QueryDispatcher>::compute
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_impl/src/lib.rs:128:9
  24:     0x7fcf6624a07f - rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr::<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_system/src/query/plumbing.rs:520:72
  25:     0x7fcf6624a07f - rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context::<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  26:     0x7fcf6624a07f - <std[64b2b1c9aa015027]::thread::local::LocalKey<core[2d39e5c12c705ebb]::cell::Cell<*const ()>>>::try_with::<rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/thread/local.rs:513:12
  27:     0x7fcf6624a07f - <std[64b2b1c9aa015027]::thread::local::LocalKey<core[2d39e5c12c705ebb]::cell::Cell<*const ()>>>::with::<rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/thread/local.rs:477:20
  28:     0x7fcf6624a07f - rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context::<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  29:     0x7fcf6624a07f - <rustc_query_impl[d22e6f0bf5601a23]::plumbing::QueryCtxt as rustc_query_system[42b0665e8e4da0bf]::query::QueryContext>::start_query::<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_impl/src/plumbing.rs:178:13
  30:     0x7fcf6624a07f - rustc_middle[dced8702efbfad36]::ty::context::tls::with_related_context::<<rustc_query_impl[d22e6f0bf5601a23]::plumbing::QueryCtxt as rustc_query_system[42b0665e8e4da0bf]::query::QueryContext>::start_query<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:112:9
  31:     0x7fcf6624a07f - rustc_middle[dced8702efbfad36]::ty::context::tls::with_context::<rustc_middle[dced8702efbfad36]::ty::context::tls::with_related_context<<rustc_query_impl[d22e6f0bf5601a23]::plumbing::QueryCtxt as rustc_query_system[42b0665e8e4da0bf]::query::QueryContext>::start_query<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:90:36
  32:     0x7fcf6624a07f - rustc_middle[dced8702efbfad36]::ty::context::tls::with_context_opt::<rustc_middle[dced8702efbfad36]::ty::context::tls::with_context<rustc_middle[dced8702efbfad36]::ty::context::tls::with_related_context<<rustc_query_impl[d22e6f0bf5601a23]::plumbing::QueryCtxt as rustc_query_system[42b0665e8e4da0bf]::query::QueryContext>::start_query<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:79:18
  33:     0x7fcf6624a07f - rustc_middle[dced8702efbfad36]::ty::context::tls::with_context::<rustc_middle[dced8702efbfad36]::ty::context::tls::with_related_context<<rustc_query_impl[d22e6f0bf5601a23]::plumbing::QueryCtxt as rustc_query_system[42b0665e8e4da0bf]::query::QueryContext>::start_query<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:90:5
  34:     0x7fcf6624a07f - rustc_middle[dced8702efbfad36]::ty::context::tls::with_related_context::<<rustc_query_impl[d22e6f0bf5601a23]::plumbing::QueryCtxt as rustc_query_system[42b0665e8e4da0bf]::query::QueryContext>::start_query<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:103:5
  35:     0x7fcf6624a07f - <rustc_query_impl[d22e6f0bf5601a23]::plumbing::QueryCtxt as rustc_query_system[42b0665e8e4da0bf]::query::QueryContext>::start_query::<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_impl/src/plumbing.rs:162:9
  36:     0x7fcf6624a07f - rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr::<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_system/src/query/plumbing.rs:520:22
  37:     0x7fcf6624a07f - rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job::<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>, false>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_system/src/query/plumbing.rs:457:9
  38:     0x7fcf6624a07f - rustc_query_system[42b0665e8e4da0bf]::query::plumbing::try_execute_query::<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>, false>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_system/src/query/plumbing.rs:401:13
  39:     0x7fcf664378b1 - rustc_query_system[42b0665e8e4da0bf]::query::plumbing::get_query_non_incr::<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_system/src/query/plumbing.rs:837:32
  40:     0x7fcf664378b1 - stacker[9cf2d74396c114e3]::maybe_grow::<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::get_query_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  41:     0x7fcf664378b1 - rustc_data_structures[2e7263bb15eb1022]::stack::ensure_sufficient_stack::<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::query::plumbing::get_query_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>::{closure#0}>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_data_structures/src/stack.rs:21:5
  42:     0x7fcf664378b1 - rustc_query_system[42b0665e8e4da0bf]::query::plumbing::get_query_non_incr::<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: true }}>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_system/src/query/plumbing.rs:837:5
  43:     0x7fcf664378b1 - rustc_query_impl[d22e6f0bf5601a23]::query_impl::codegen_fn_attrs::get_query_non_incr::__rust_end_short_backtrace
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_impl/src/plumbing.rs:617:26
  44:     0x7fcf6550b185 - rustc_middle[dced8702efbfad36]::query::inner::query_get_at::<rustc_query_system[42b0665e8e4da0bf]::query::caches::DefIdCache<rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/query/inner.rs:28:17
  45:     0x7fcf6550b185 - <rustc_middle[dced8702efbfad36]::query::plumbing::TyCtxtAt>::codegen_fn_attrs::<rustc_span[5b13d0a724d2c192]::def_id::DefId>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/query/plumbing.rs:429:42
  46:     0x7fcf6550b185 - <rustc_middle[dced8702efbfad36]::ty::context::TyCtxt>::codegen_fn_attrs::<rustc_span[5b13d0a724d2c192]::def_id::DefId>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/query/plumbing.rs:418:35
  47:     0x7fcf6550b185 - <rustc_middle[dced8702efbfad36]::ty::context::TyCtxt>::adjust_target_feature_sig
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context.rs:2108:37
  48:     0x7fcf64981695 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::sig_for_fn_def_coercion
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/coercion.rs:1113:23
  49:     0x7fcf64890e50 - <rustc_hir_typeck[d279b0886461d6fe]::coercion::Coerce>::coerce_from_fn_item
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/coercion.rs:876:34
  50:     0x7fcf64890e50 - <rustc_hir_typeck[d279b0886461d6fe]::coercion::Coerce>::coerce::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/coercion.rs:293:22
  51:     0x7fcf64a1bab3 - <rustc_hir_typeck[d279b0886461d6fe]::coercion::Coerce>::coerce
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/coercion.rs:221:5
  52:     0x7fcf64982717 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::coerce::{closure#2}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/coercion.rs:988:47
  53:     0x7fcf64982717 - <rustc_infer[4c50fa8b53be22ad]::infer::InferCtxt>::commit_if_ok::<rustc_infer[4c50fa8b53be22ad]::infer::InferOk<(alloc[21b5fb6b1b472667]::vec::Vec<rustc_middle[dced8702efbfad36]::ty::adjustment::Adjustment>, rustc_middle[dced8702efbfad36]::ty::Ty)>, rustc_type_ir[330d6fd2821dabe]::error::TypeError<rustc_middle[dced8702efbfad36]::ty::context::TyCtxt>, <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::coerce::{closure#2}>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_infer/src/infer/snapshot/mod.rs:86:17
  54:     0x7fcf64982717 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::coerce
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/coercion.rs:988:23
  55:     0x7fcf648edbd6 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::demand_coerce_diag
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/demand.rs:269:28
  56:     0x7fcf64952fb0 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type_or_error::<<rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type::{closure#0}>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:164:20
  57:     0x7fcf64952fb0 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:152:14
  58:     0x7fcf64952fb0 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_decl_initializer
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:897:18
  59:     0x7fcf64952fb0 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_decl
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:907:32
  60:     0x7fcf649582eb - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_decl_local
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:942:23
  61:     0x7fcf649582eb - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_stmt
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:966:22
  62:     0x7fcf649582eb - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_block::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1043:22
  63:     0x7fcf649582eb - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::with_breakable_ctxt::<<rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_block::{closure#0}, ()>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs:1539:22
  64:     0x7fcf649582eb - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_block
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs:1041:31
  65:     0x7fcf648b27b8 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_kind
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:387:46
  66:     0x7fcf648e3267 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:291:23
  67:     0x7fcf648e3267 - stacker[9cf2d74396c114e3]::maybe_grow::<rustc_middle[dced8702efbfad36]::ty::Ty, <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /rust/deps/stacker-0.1.21/src/lib.rs:57:9
  68:     0x7fcf648e3267 - rustc_data_structures[2e7263bb15eb1022]::stack::ensure_sufficient_stack::<rustc_middle[dced8702efbfad36]::ty::Ty, <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args::{closure#0}>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_data_structures/src/stack.rs:21:5
  69:     0x7fcf648e3267 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:286:18
  70:     0x7fcf648e084c - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_with_expectation
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:231:14
  71:     0x7fcf64878970 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_with_hint
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:186:14
  72:     0x7fcf64878970 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type_or_error::<<rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type::{closure#0}>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:162:23
  73:     0x7fcf64878970 - <rustc_hir_typeck[d279b0886461d6fe]::fn_ctxt::FnCtxt>::check_expr_coercible_to_type
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/expr.rs:152:14
  74:     0x7fcf64878970 - rustc_hir_typeck[d279b0886461d6fe]::typeck_with_inspect::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/lib.rs:209:13
  75:     0x7fcf6489f4ec - rustc_hir_typeck[d279b0886461d6fe]::typeck_with_inspect
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_hir_typeck/src/lib.rs:103:1
  76:     0x7fcf664bf89f - rustc_query_impl[d22e6f0bf5601a23]::query_impl::typeck::compute_fn::__rust_begin_short_backtrace
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_impl/src/plumbing.rs:290:9
  77:     0x7fcf662f6c55 - <rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_data_structures[2e7263bb15eb1022]::vec_cache::VecCache<rustc_hir_id[e6d3f9673b591e6d]::OwnerId, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::dep_graph::graph::DepNodeIndex>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}> as rustc_query_system[42b0665e8e4da0bf]::query::dispatcher::QueryDispatcher>::compute
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_impl/src/lib.rs:128:9
  78:     0x7fcf662f6c55 - rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr::<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_data_structures[2e7263bb15eb1022]::vec_cache::VecCache<rustc_hir_id[e6d3f9673b591e6d]::OwnerId, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::dep_graph::graph::DepNodeIndex>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_query_system/src/query/plumbing.rs:520:72
  79:     0x7fcf662f6c55 - rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context::<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_data_structures[2e7263bb15eb1022]::vec_cache::VecCache<rustc_hir_id[e6d3f9673b591e6d]::OwnerId, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::dep_graph::graph::DepNodeIndex>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:60:9
  80:     0x7fcf662f6c55 - <std[64b2b1c9aa015027]::thread::local::LocalKey<core[2d39e5c12c705ebb]::cell::Cell<*const ()>>>::try_with::<rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_data_structures[2e7263bb15eb1022]::vec_cache::VecCache<rustc_hir_id[e6d3f9673b591e6d]::OwnerId, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::dep_graph::graph::DepNodeIndex>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/thread/local.rs:513:12
  81:     0x7fcf662f6c55 - <std[64b2b1c9aa015027]::thread::local::LocalKey<core[2d39e5c12c705ebb]::cell::Cell<*const ()>>>::with::<rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_data_structures[2e7263bb15eb1022]::vec_cache::VecCache<rustc_hir_id[e6d3f9673b591e6d]::OwnerId, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::dep_graph::graph::DepNodeIndex>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/thread/local.rs:477:20
  82:     0x7fcf662f6c55 - rustc_middle[dced8702efbfad36]::ty::context::tls::enter_context::<rustc_query_system[42b0665e8e4da0bf]::query::plumbing::execute_job_non_incr<rustc_query_impl[d22e6f0bf5601a23]::SemiDynamicQueryDispatcher<rustc_data_structures[2e7263bb15eb1022]::vec_cache::VecCache<rustc_hir_id[e6d3f9673b591e6d]::OwnerId, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>, rustc_query_system[42b0665e8e4da0bf]::dep_graph::graph::DepNodeIndex>, {rustc_query_impl[d22e6f0bf5601a23]::QueryFlags { is_anon: false, is_depth_limit: false, is_feedable: false }}>>::{closure#0}, rustc_middle[dced8702efbfad36]::query::erase::ErasedData<[u8; 8usize]>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/compiler/rustc_middle/src/ty/context/tls.rs:57:9
  

...



 273:     0x7fcf63f295c5 - std[64b2b1c9aa015027]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/sys/backtrace.rs:166:18
 274:     0x7fcf63f4be03 - std[64b2b1c9aa015027]::thread::lifecycle::spawn_unchecked::<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/thread/lifecycle.rs:91:13
 275:     0x7fcf63f4be03 - <core[2d39e5c12c705ebb]::panic::unwind_safe::AssertUnwindSafe<std[64b2b1c9aa015027]::thread::lifecycle::spawn_unchecked<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}> as core[2d39e5c12c705ebb]::ops::function::FnOnce<()>>::call_once
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/core/src/panic/unwind_safe.rs:274:9
 276:     0x7fcf63f4be03 - std[64b2b1c9aa015027]::panicking::catch_unwind::do_call::<core[2d39e5c12c705ebb]::panic::unwind_safe::AssertUnwindSafe<std[64b2b1c9aa015027]::thread::lifecycle::spawn_unchecked<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:581:40
 277:     0x7fcf63f4be03 - std[64b2b1c9aa015027]::panicking::catch_unwind::<(), core[2d39e5c12c705ebb]::panic::unwind_safe::AssertUnwindSafe<std[64b2b1c9aa015027]::thread::lifecycle::spawn_unchecked<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panicking.rs:544:19
 278:     0x7fcf63f4be03 - std[64b2b1c9aa015027]::panic::catch_unwind::<core[2d39e5c12c705ebb]::panic::unwind_safe::AssertUnwindSafe<std[64b2b1c9aa015027]::thread::lifecycle::spawn_unchecked<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}::{closure#0}>, ()>
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/panic.rs:359:14
 279:     0x7fcf63f4be03 - std[64b2b1c9aa015027]::thread::lifecycle::spawn_unchecked::<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/thread/lifecycle.rs:89:26
 280:     0x7fcf63f4be03 - <std[64b2b1c9aa015027]::thread::lifecycle::spawn_unchecked<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_with_globals<rustc_interface[e59c08b5eb50d667]::util::run_in_thread_pool_with_globals<rustc_interface[e59c08b5eb50d667]::interface::run_compiler<(), rustc_driver_impl[ecb1fe8f07c57781]::run_compiler::{closure#0}>::{closure#1}, ()>::{closure#0}, ()>::{closure#0}::{closure#0}, ()>::{closure#1} as core[2d39e5c12c705ebb]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
                               at /rustc-dev/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/core/src/ops/function.rs:250:5
 281:     0x7fcf674e12ef - <alloc[21b5fb6b1b472667]::boxed::Box<dyn core[2d39e5c12c705ebb]::ops::function::FnOnce<(), Output = ()> + core[2d39e5c12c705ebb]::marker::Send> as core[2d39e5c12c705ebb]::ops::function::FnOnce<()>>::call_once
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/alloc/src/boxed.rs:2206:9
 282:     0x7fcf674e12ef - <std[64b2b1c9aa015027]::sys::thread::unix::Thread>::new::thread_start
                               at /rustc/13c38730d981289cc7ae4cc109fd7756bf83ee67/library/std/src/sys/thread/unix.rs:118:17
 283:     0x7fcf6109698b - <unknown>
 284:     0x7fcf6111a9cc - <unknown>
 285:                0x0 - <unknown>

error: the compiler unexpectedly panicked. This is a bug

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please make sure that you have updated to the latest nightly

note: rustc 1.95.0-nightly (13c38730d 2026-02-08) running on x86_64-unknown-linux-gnu

query stack during panic:
#0 [codegen_fn_attrs] computing codegen attributes of `SSE2_ONLY`
#1 [typeck] type-checking `SSE2_ONLY`
#2 [check_match] match-checking `SSE2_ONLY`
#3 [mir_built] building MIR for `SSE2_ONLY`
#4 [trivial_const] checking if `SSE2_ONLY` is a trivial const
#5 [eval_to_const_value_raw] simplifying constant for the type system `SSE2_ONLY`
#6 [analysis] running analysis passes on crate `mvce`
end of query stack
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0601`.

```

</p>
</details>

--- a/compiler/rustc_middle/src/ty/context.rs
+++ b/compiler/rustc_middle/src/ty/context.rs
@@ -1330,8 +1330,8 @@ impl<'tcx> TyCtxt<'tcx> {
         caller: DefId,
     ) -> Option<ty::Binder<'tcx, ty::FnSig<'tcx>>> {
         let fun_features = &self.codegen_fn_attrs(fun_def).target_features;
-        let callee_features = &self.codegen_fn_attrs(caller).target_features;
-        if self.is_target_feature_call_safe(&fun_features, &callee_features) {
+        let caller_features = &self.body_codegen_attrs(caller).target_features;
+        if self.is_target_feature_call_safe(&fun_features, &caller_features) {
             return Some(fun_sig.map_bound(|sig| ty::FnSig { safety: hir::Safety::Safe, ..sig }));
         }
         None

--- a/tests/ui/target-feature/const-target-feature-fn-ptr-coercion.rs
+++ b/tests/ui/target-feature/const-target-feature-fn-ptr-coercion.rs
@@ -1,0 +1,33 @@
+//@ only-x86_64
+//@ check-pass
+//
+// Regression test for <https://github.com/rust-lang/rust/issues/152340>.
+
+#![allow(dead_code)]
+
+#[target_feature(enable = "sse2")]
+const fn foo() {}
+
+// DefKind::Const
+const _: () = unsafe {
+    let _: unsafe fn() = foo;
+};
+
+// DefKind::AssocConst
+struct S;
+impl S {
+    const C: () = unsafe {
+        let _: unsafe fn() = foo;
+    };
+}
+
+// DefKind::InlineConst
+fn bar() {
+    let _ = const {
+        unsafe {
+            let _: unsafe fn() = foo;
+        }
+    };
+}
+
+fn main() {}

